### ISSUE TITLE
Add deep_copy to profiling tools

### ISSUE THREAD

**Issue by stanmoore1** [kokkos/kokkos-tools#49]
Can we please add `Kokkos::deep_copy` to the `kernel_logger` and `memory_events` tools?

**Comment by crtrott** [comment]
Hm it should be in there, but you are right I also had the feeling it didn't log all of them.

**Comment by stanmoore1** [comment]
@crtrott, these tools don't log *any* deep_copy events:

```c++
#include <Kokkos_Core.hpp>
#include <cstdio>

typedef Kokkos::View<int*> view_type;

int main (int argc, char* argv[]) {
  Kokkos::initialize (argc, argv);
  {
    const int N = 10;

    view_type a (Kokkos::view_alloc("A",Kokkos::WithoutInitializing), N);
    Kokkos::deep_copy(a,7);

    view_type b (Kokkos::view_alloc("B",Kokkos::WithoutInitializing), N);
    Kokkos::deep_copy(b,a);

    auto c = Kokkos::create_mirror_view(a);
    Kokkos::deep_copy(c,a);

  }
  Kokkos::finalize ();
}
```
Kernel logger:
```
KokkosP: Allocate<Cuda> name: A pointer: 0x7fff7a200080 size: 40
KokkosP: Executing parallel-for kernel on device 0 with unique execution identifier 0
KokkosP: Kokkos::ViewFill-1D
KokkosP: Execution of kernel 0 is completed.
KokkosP: Allocate<Cuda> name: B pointer: 0x7fff7a200280 size: 40
KokkosP: Allocate<Host> name: A_mirror pointer: 0x4c5aa380 size: 40
KokkosP: Executing parallel-for kernel on device 0 with unique execution identifier 1
KokkosP: Kokkos::View::initialization
KokkosP: Execution of kernel 1 is completed.
KokkosP: Deallocate<Host> name: A_mirror pointer: 0x4c5aa380 size: 40
KokkosP: Deallocate<Cuda> name: B pointer: 0x7fff7a200280 size: 40
KokkosP: Deallocate<Cuda> name: A pointer: 0x7fff7a200080 size: 40
KokkosP: Kokkos library finalization called.
```
Memory events:
```
0.460472   0x7fff5a200080             40             Cuda Allocate   A
0.460597   0x7fff5a200280             40             Cuda Allocate   B
0.460695       0x1ae3ad40             40             Host Allocate   A_mirror
```

**Comment by stanmoore1** [comment]
Fixed by kokkos/kokkos#2523.
